server/src/internalClusterTest/java/org/opensearch/search/aggregations/metrics/SumLongIT.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.search.aggregations.metrics;
+
+import org.opensearch.action.index.IndexRequestBuilder;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.opensearch.index.query.QueryBuilders.matchAllQuery;
+import static org.opensearch.search.aggregations.AggregationBuilders.sum;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertHitCount;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class SumLongIT extends OpenSearchIntegTestCase {
+    public void testLargeLongValues() throws InterruptedException {
+        prepareCreate("longidx").setMapping("value", "type=long").get();
+
+        List<IndexRequestBuilder> builders = new ArrayList<>();
+        builders.add(client().prepareIndex("longidx").setSource("value", -922337202685477588L));
+        builders.add(client().prepareIndex("longidx").setSource("value", 922337202685477587L));
+        builders.add(client().prepareIndex("longidx").setSource("value", 1000L));
+        builders.add(client().prepareIndex("longidx").setSource("value", -500L));
+
+        indexRandom(true, builders);
+        ensureSearchable();
+
+        SearchResponse searchResponse = client().prepareSearch("longidx")
+            .setQuery(matchAllQuery())
+            .addAggregation(sum("sum").field("value"))
```

Comment:
I don't see where we're setting the "method". 
We should also consider writing a test case for "method=kahan" to show that these aggregations get approximate results.

Reply:
updated and more tests added in https://github.com/opensearch-project/OpenSearch/pull/6984/commits/7d50f500d045f1c8acb3d71318ef777cef6e6143